torch/distributed/distributed_c10d.py

```diff
@@ -347,15 +362,14 @@ def init_process_group(backend,
         raise RuntimeError("trying to initialize the default process group "
                            "twice!")
 
-    world_size = kwargs.pop('world_size', -1)
-    group_name = kwargs.pop('group_name', '')
-    rank = kwargs.pop('rank', -1)
-    store = kwargs.pop('store', None)
+    assert (store is None) or (init_method is None), \
+        "Cannot specify both init_method and store."
+
     if store is not None:
-        assert world_size > 0, 'world_size needs to be positive'
-        assert rank >= 0, 'rank needs to be non-negative'
-    assert len(kwargs) == 0, \
-        "got unexpected keyword arguments: %s" % ",".join(kwargs.keys())
+        assert world_size > 0, 'world_size must be positive if using store'
+        assert rank >= 0, 'rank must be non-negative if using store'
```

Comment:
nit: shall we mention this in the doc?

Reply:
Isn't it safe to think this is implied?

Reply:
Ah, I see these asserts fire if you don't set them at all, and they are set to their default value of `-1`. I think the assert should reflect this and mention that the caller must specify both `world_size` and `rank` keyword arguments to the `init_process_group` call. Then they'll know what to do. In isolation they are not that helpful (if you don't know about those kwargs).

Reply:
@mrshenli - just want to double check, the comments in this function explain the conditions under which world_size and rank are required (they aren't required if you pass in a URL with them encoded already). If that's good enough I'll resolve this conversation.

Reply:
@kuttas yes, that's good enough. :)